### PR TITLE
[Resolver] Allow platform detection to work for nested dependencies

### LINKED ISSUE #4084: pod lint uses incompatible dependency version
My podspec specifies platform ```:ios, '6.0'``` and a dependency to AFNetworking ```~> 2.5```. When i reference this as a development pod the best compatible version of AFNetworking gets installed, which is 2.5.4 because 2.6 requires iOS 7. But when i lint this podspec i get the following notice

    [!] The platform of the target `Pods` (iOS 6.0) may not be compatible with `AFNetworking (2.6.0)` which has a minimum requirement of iOS 7.0 - OS X 10.9 - watchOS 2.0.

Obviously an incompatible version of AFNetworking is beeing used during lint.

### PR DESCRIPTION
**Pull request by segiddins**
Closes #4084.

- [x] Test
- [x] CHANGELOG

### CODE REVIEW

File: lib/cocoapods/installer/analyzer.rb [hunk 1]
```diff
@@ -532,7 +532,7 @@ def resolve_dependencies
         end
 
         specs_by_target = nil
-        UI.section "Resolving dependencies of #{UI.path podfile.defined_in_file}" do
+        UI.section "Resolving dependencies of #{UI.path(podfile.defined_in_file) || 'Podfile'}" do
```
Comment (kylef) [inline lib/cocoapods/installer/analyzer.rb]: This change seems unrelated? Also, shouldn't `podfile.defined_in_file` always have a value?

Comment (segiddins) [inline lib/cocoapods/installer/analyzer.rb]: Not when linting

Comment (segiddins) [inline lib/cocoapods/installer/analyzer.rb]: (Which is what prompted this PR, so I made that change whilst debugging)

Comment (kylef) [inline lib/cocoapods/installer/analyzer.rb]: :white_check_mark:

**Comment by segiddins** [comment]
Please review @kylef , this should be mergable.

**Comment by kylef** [comment]
Good, but it needs to be :green_apple: green.

**Comment by segiddins** [comment]
Will be green once https://github.com/CocoaPods/CocoaPods/pull/4104#issuecomment-136759535 is merged
